src/test/test_step.go

```diff
@@ -366,6 +366,9 @@ func doTestResults(tid int, state *core.BuildState, target *core.BuildTarget, ru
 
 	if runRemotely {
 		metadata, err = state.RemoteClient.Test(tid, target, run)
+		if metadata == nil {
```

Comment:
Should we just return a non-nil metadata from `Test`?

Reply:
Using the return value of something that errored seems like an anti-pattern no?